### PR TITLE
Update elastic search spy to check for new gem name

### PR DESCRIPTION
**Pull request by jaggederest**
## What does this pull request do?
There is an error in the tests, and likely no data being captured, with the new elasticsearch-ruby gem. This fixes that issue.

## Why is it important?
elasticsearch-ruby relies on a separate gem to provide transport functionality, and that gem is what we instrument. Last year, in preparation for the 8.0 release, that gem was renamed and separated from the core elasticsearch-ruby repo into a new repo, and renamed elastic-transport. Our spy still works, but it was no longer checking the right dependency name to install itself, and therefore was not capturing data correctly.

## Checklist

- [x] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/).
- [x] My code follows the style guidelines of this project (See `.rubocop.yml`)
- [x] I have rebased my changes on top of the latest main branch

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing [**unit** tests](https://github.com/elastic/apm-agent-ruby/blob/main/CONTRIBUTING.md#testing) pass locally with my changes

- [x] I have made corresponding changes to the documentation
- [x] I have updated [CHANGELOG.asciidoc](CHANGELOG.asciidoc)
- [x] I have updated [supported-technologies.asciidoc](docs/supported-technologies.asciidoc)
- [x] Added an API method or config option? Document in which version this will be introduced

### CODE REVIEW

**Review by estolfo — APPROVED** [review]

File: lib/elastic_apm/spies/elasticsearch.rb [hunk 1]
```diff
@@ -79,7 +79,13 @@ def perform_request(method, path, *args, &block)
       end
 
       def install
-        ::Elasticsearch::Transport::Client.prepend(Ext)
+        if defined?(::ElasticSearch::Transport::Client)
```
[the remark this answers is not in the document]
Comment (jaggederest) [reply lib/elastic_apm/spies/elasticsearch.rb:82]: There are 3 different constants in use through the versions of elastic transport ruby - `Elastic::Transport` is the current one, the older one is `ElasticSearch::Transport`, and the oldest one is `Elasticsearch::Transport`. So this if statement makes sure we instrument the correct one of those three, based on which one is defined. I should probably make it `Elastic::Transport` first, now that I think about it 👍

File: Gemfile [hunk 2]
```diff
@@ -46,18 +46,17 @@ if !defined?(JRUBY_VERSION) && RUBY_VERSION < '2.5'
   gem 'google-protobuf', '< 3.12'
 end
 gem 'grpc' if !defined?(JRUBY_VERSION) && RUBY_VERSION < '3.0'
-gem 'json'
+gem 'json', '2.5.1'
```
Comment (jaggederest) [inline Gemfile:49]: Needs a note that 2.6.1 has a segfault with ruby >=3.0.4, can be unpinned after a new release fixes that issue

File: Gemfile [hunk 3]
```diff
@@ -116,6 +115,11 @@ else
   gem 'sqlite3'
 end
 
+# current sneakers only supports >=2.5.0
+if Gem::Version.create(RUBY_VERSION) >= Gem::Version.create('2.5.0')
```
Comment (jaggederest) [inline Gemfile:119]: Should move to a separate PR with all gem changes

File: spec/elastic_apm/config/wildcard_pattern_list_spec.rb [hunk 4]
```diff
@@ -23,9 +23,9 @@ module ElasticAPM
   class Config
     RSpec.describe(WildcardPatternList::WildcardPattern) do
       subject { described_class.new(pattern) }
-
+      json_file = File.read("spec/fixtures/wildcard_matcher_tests.json", :encoding => 'utf-8')
```
Comment (jaggederest) [inline spec/elastic_apm/config/wildcard_pattern_list_spec.rb:26]: Can return this to single-line format, move to gem update PR

File: spec/elastic_apm/spies/sneakers_spec.rb [hunk 5]
```diff
@@ -18,93 +18,94 @@
...
-
-require 'sneakers'
-require 'elastic_apm/spies/sneakers'
-
-module ElasticAPM
-  RSpec.describe 'Spy: Sneakers', :intercept do
-    class MockQueue
-      def name
-        'q1'
+unless Gem::Specification.find_all_by_name('sneakers', '>=2.12.0').empty?
```
Comment (jaggederest) [inline spec/elastic_apm/spies/sneakers_spec.rb:21]: Gem update PR

**Comment by jaggederest** [comment]
@estolfo This PR should be ready for re-review, thanks for reviewing it!

File: lib/elastic_apm/spies/elasticsearch.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (estolfo) [reply lib/elastic_apm/spies/elasticsearch.rb:82]: Where did you find that it was `ElasticSearch` previously?

[the remark this answers is not in the document]
Comment (jaggederest) [reply lib/elastic_apm/spies/elasticsearch.rb:82]: Huh, maybe it wasn't. That was just the constant I used when I fixed the test errors. I will remove it and if it doesn't cause any additional test errors, no harm no foul I guess.

**Review by estolfo — APPROVED** [review]
